CMakeLists.txt

```diff
@@ -253,7 +254,7 @@ option(QUDA_FAST_COMPILE_DSLASH "enable fast compilation in dslash kernels (~20%
 
 option(QUDA_OPENMP "enable OpenMP" OFF)
 set(QUDA_CXX_STANDARD
-    14
+    17
```

Comment:
This leads to CMake failure on my system (CMake 3.16.3, gcc 9.3.0, cuda 11.0.194). Setting `-DQUDA_CXX_STANDARD=14` works around the issue.

Is this expected?

```
-- The CUDA compiler identification is NVIDIA 11.0.194
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc
CMake Error in /home/bartek/build/quda-ndeg_twisted_clover/CMakeFiles/CMakeTmp/CMakeLists.txt:
  Target "cmTC_fc08a" requires the language dialect "CUDA17" , but CMake does
  not know the compile flags to use to enable it.


CMake Error at /usr/share/cmake-3.16/Modules/CMakeTestCUDACompiler.cmake:30 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  CMakeLists.txt:463 (enable_language)
```

Reply:
I encountered that, it's due to CMake 3.16.3 not being able to handle C++ 17 standard. If you bump up to 3.18.x it goes away.

Reply:
Yes, this was added in 3.18: See https://cmake.org/cmake/help/v3.18/prop_tgt/CUDA_STANDARD.html#prop_tgt:CUDA_STANDARD 
https://cmake.org/cmake/help/v3.17/prop_tgt/CUDA_STANDARD.html#prop_tgt:CUDA_STANDARD

We need to bump up the required cmake version to 3.18 which we will need to do anyway for https://github.com/lattice/quda/pull/1120

Reply:
Thanks! Now how to convince admins to provide versions of software less than two or three years old ;) (without having to manually install it)